torch/_dynamo/guards.py

```diff
@@ -569,11 +570,11 @@ def get_sources(t_id, dim):
         for shape_guard in guards:
             self._produce_guard_code(guard, [shape_guard], shape_env=True)
 
-    def TENSOR_MATCH(self, guard: Guard):
+    def TENSOR_MATCH(self, guard: Guard, value=None):
```

Comment:
if .get() invokes an fn, like in the source added in this PR (NumpyTensorSource) - we do not have id stability, which is required between example values from `get` and traced values at fake-ification time, because of how we build the sizes/strides for tensor guards.

Reply:
I think I really need a comment here explaining what the value arg does. Still not sure from the comment either, need to read more.

Reply:
okay, will add a comment.